src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -265,197 +280,168 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
-                    {
-                        reducedMethodSymbol = methodSymbol.ReduceExtensionMethod(receiverTypeSymbol);
-                    }
+                // By default, don't trigger index creation except for documents in current project.
+                var loadOnly = !forceIndexCreation && projectId != currentProject.Id;
+                var cacheEntry = await GetCacheEntryAsync(project, loadOnly, cacheService, cancellationToken).ConfigureAwait(false);
 
-                    if (reducedMethodSymbol != null)
-                    {
-                        var symbolKeyData = SymbolKey.CreateString(reducedMethodSymbol);
-                        builder.Add(new SerializableImportCompletionItem(
-                            symbolKeyData,
-                            reducedMethodSymbol.Name,
-                            reducedMethodSymbol.Arity,
-                            reducedMethodSymbol.GetGlyph(),
-                            qualifiedNamespaceName));
-                    }
+                if (cacheEntry == null)
+                {
+                    // Don't provide anything if we don't have all the required SyntaxTreeIndex created.
+                    return GetIndicesResult.NoneResult;
                 }
-            }
-        }
 
-        private static void GetItemsFromConflictingTypes(
-            INamespaceSymbol containingNamespaceSymbol,
-            ConflictNameNode conflictTypeNodes,
-            ArrayBuilder<SerializableImportCompletionItem> builder,
-            ITypeSymbol receiverTypeSymbol,
-            SemanticModel semanticModel,
-            int position,
-            StatisticCounter counter)
-        {
-            Debug.Assert(!conflictTypeNodes.NamespaceAndMethodNames.HasValue);
+                syntaxIndices.Add(project, cacheEntry.Value);
+            }
 
-            foreach (var child in conflictTypeNodes.Children.Values)
+            // Search through all direct PE references.
+            foreach (var peReference in currentProject.MetadataReferences.OfType<PortableExecutableReference>())
             {
-                if (child.NamespaceAndMethodNames == null)
+                var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
+                    solution, peReference, loadOnly: !forceIndexCreation, cancellationToken).ConfigureAwait(false);
+
+                if (info == null)
                 {
-                    var childNamespace = containingNamespaceSymbol.GetMembers(child.Name).OfType<INamespaceSymbol>().FirstOrDefault();
-                    if (childNamespace != null)
-                    {
-                        GetItemsFromConflictingTypes(childNamespace, child, builder, receiverTypeSymbol, semanticModel, position, counter);
-                    }
+                    // Don't provide anything if we don't have all the required SymbolTreeInfo created.
+                    return GetIndicesResult.NoneResult;
```

Comment:
this seems really sucky.  so i'm unable to add extensions for things i *do* know about, jsut because we don't know everything?  i would really change this all to be best effort.  return what you do have.

Reply:
I think this is OK, since the expander button is still available when we hit this path (as unselected). And when user explicitly click the expander, we will force the index creation and wait for it to finish.

Reply:
@CyrusNajmabadi ?

Reply:
still dont' like it.  seems unnecessary and is code that only exists to degrade the user experience.  i still don't have a good idea of why we don't just skip this index and move onto the next one...

Reply:
I don't agree. I think returning a different collection of unimported extension methods for a given object based on the availability of indices is more of an UX degrade, i.e. it makes the behavior unpredictable in the sense that you never know for sure if all the ext methods are returned (well, supported one at least, since we don't show all of them, e.g. internals from PE references). Say if someone hit this case and couldn't find an existing ext method they are looking for, they might think it doesn't exist. Expecting them to try to trigger the completion again as a way to correct the issue in my opinion would be a leak of implementation detail, or worse, it gives users the impression this might be a bug.

In contrast, not providing unimported extension methods if we don't have all the info, and show expander icon as unselected, has the advantage of clarity. Most of the time when someone hit this, they are likely not even looking for ext methods so this would be ignored, whereas when it happens when they do, they would know for sure that unimported ext method is not included in the list, and can be added by clicking expander.

And I'd like to emphasis this again, this only happens when the indices of references are not fully constructed, which shouldn't be very common scenario (usually created by solution crawler, but using expander, as well as other index back features would force the construction)

Reply:
Synced offline. Seems we have regressed completion's "partial semantic" behavior with async completion, which would return incomplete result if solution isn't fully loaded. IN that case, it makes sense to make import completion stick to this behavior.

Reply:
i disagree still :)

note: this isn't a hypothetical.  async completion returns everything int he normal symbol system.  but throws out the import completoin stuff.  i hit this routinely and i hate it :)